### PR TITLE
[prometheus.operator.probes] Add namespace label with the namespace of the probe object

### LINKED ISSUE #1049: [prometheus.operator.probes] add namespace label
### What's wrong?

When using the prometheus probe scraper, the namespace label is added to the scraped metrics.
When using the alloy component named prometheus.operator.probes to scrape probes, this label is not accessible.
In the debug/info the namespace is listed, so the data is there but not added as a label in the metrics.

I tried to add a rule to add the label. 
```
rule {
          source_labels = ["__meta_kubernetes_namespace"]
          target_label = "namespace"
        }
```
But without any succes.

could it be possible to add the label via relabeling or via code to the metrics?

### Steps to reproduce

Add a probe in kuberentes and use the alloy component to scrape the probe. look for the namespace label.

### System information

Kubernetes

### Software version

Alloy v1.1.1

### Configuration

_No response_

### Logs

_No response_

### PR DESCRIPTION
**Pull request by ToonTijtgat2**
#### PR Description
Add namespace label with the namespace of the probe object in the static config to align with prometheus setup.
Prometheus automatically adds the namespace label, and since this is standard practice and we rely on this label for some dashboards and rules, we'd like to align the prometheus setup and alloy setup.

#### Which issue(s) this PR fixes

Fixes #1049 

#### PR Checklist

- [x] CHANGELOG.md updated
- [ ] Documentation added
- [x] Tests updated
- [ ] Config converters updated

### CODE REVIEW

**Comment by ToonTijtgat2** [comment]
@thampiotr  I started a new PR for this since the previous is all broken.
Thanks for checking it here

**Comment by ToonTijtgat2** [comment]
@thampiotr  is all oke for this PR, or is there something I can do to help?

**Comment by ToonTijtgat2** [comment]
@thampiotr nice, all checks past. :)

**Comment by ToonTijtgat2** [comment]
@thampiotr thanks for launching the last continuous-integration/drone/pr test, since it succeeded, does this mean all is oke for the PR to be merged?

**Comment by thampiotr** [comment]
> @thampiotr thanks for launching the last continuous-integration/drone/pr test, since it succeeded, does this mean all is oke for the PR to be merged?

It means that the tests didn't detect any issues, next step is human review. I'll try to review it in more detail soon.

File: internal/component/prometheus/operator/configgen/config_gen_probe.go [hunk 1]
```diff
@@ -71,6 +71,8 @@ func (cg *ConfigGenerator) GenerateProbeConfig(m *promopv1.Probe) (cfg *config.S
 		for k, v := range static.Labels {
 			grp.Labels[model.LabelName(k)] = model.LabelValue(v)
 		}
+		// Add namespace label
+		grp.Labels["namespace"] = model.LabelValue(m.Namespace)
```
Comment (thampiotr) [inline internal/component/prometheus/operator/configgen/config_gen_probe.go:75]:
```suggestion
		// Add a namespace label if not already set. See https://github.com/prometheus-operator/prometheus-operator/commit/ffce8d01ff798135d5bbfcba1a88a87a418ee9fc.
		if _, ok := grp.Labels["namespace"]; !ok {
			grp.Labels["namespace"] = model.LabelValue(m.Namespace)
		}
```
The label should be set only if it wasn't already set by the the static labels. See the upstream code for details.

File: CHANGELOG.md [hunk 2]
```diff
@@ -18,6 +18,7 @@ Main (unreleased)
 
 - Added a success rate panel on the Prometheus Components dashboard. (@thampiotr)
 - Add namespace field to Faro payload (@cedricziel)
+- Add namespace label with the namespace of the probe object in the static config to align with prometheus setup. (@toontijtgat2)
```
Comment (thampiotr) [inline CHANGELOG.md:21]:
```suggestion

- Added a `namespace` label to probes scraped by the `prometheus.operator.probes` component to align with the upstream Prometheus Operator setup. (@toontijtgat2)
```

**Review by thampiotr — COMMENTED** [review]
@ToonTijtgat2 thank you for the contribution! I have a couple of suggestions, but it should go smooth from here.

**Comment by ToonTijtgat2** [comment]
@thampiotr Thanks for the suggestions! I commited them.

**Review by thampiotr — APPROVED** [review]

**Pull request merged**